artichoke-backend/src/intern.rs

```diff
@@ -11,61 +10,9 @@ use crate::ffi::InterpreterExtractError;
 use crate::sys;
 use crate::Artichoke;
 
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Symbol(intaglio::Symbol);
```

Comment:
there were 3 different `Symbol` wrappers around `u32` floating around in `artichoke-backend`:

- this one.
- `Symbol` in `extn::core::symbol` module.
- `intaglio::Symbol`.

One of the nice simplifications of this PR is replacing all of the Ruby Core pieces with `spinoso_symbol::Symbol` and replacing the rest of the internals with `u32`, only marshaling to the `intaglio` repr at the edge.